File: app/src/main/java/com/example/newchatui/dialogFlowHelper/RequestJavaV2Task.java
package com.example.newchatui.dialogFlowHelper;

import android.app.Activity;
import android.os.AsyncTask;

import com.example.newchatui.ui.chat.awarness.CrimePrediction;
import com.google.cloud.dialogflow.v2.DetectIntentRequest;
import com.google.cloud.dialogflow.v2.DetectIntentResponse;
import com.google.cloud.dialogflow.v2.QueryInput;
import com.google.cloud.dialogflow.v2.SessionName;
import com.google.cloud.dialogflow.v2.SessionsClient;

public class RequestJavaV2Task extends AsyncTask<Void, Void, DetectIntentResponse> {

    Activity activity;
    private SessionName session;
    private SessionsClient sessionsClient;
    private QueryInput queryInput;

    public RequestJavaV2Task(Activity activity, SessionName session, SessionsClient sessionsClient, QueryInput queryInput) {
        this.activity = activity;
        this.session = session;
        this.sessionsClient = sessionsClient;
        this.queryInput = queryInput;
    }

    @Override
    protected DetectIntentResponse doInBackground(Void... voids) {
        try{
            DetectIntentRequest detectIntentRequest =
                    DetectIntentRequest.newBuilder()
                            .setSession(session.toString())
                            .setQueryInput(queryInput)
                            .build();
            return sessionsClient.detectIntent(detectIntentRequest);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }

    @Override
    protected void onPostExecute(DetectIntentResponse response) {
        ((CrimePrediction) activity).callbackV2(response);
    }
}

File: app/src/main/java/com/example/copsbotuserapp/UI/RequestJavaV2Task.kt
package com.example.copsbotuserapp.UI

import android.app.Activity
import android.os.AsyncTask
import com.example.copsbotuserapp.MainActivity
import com.google.cloud.dialogflow.v2.*


class RequestJavaV2Task internal constructor(
    var activity: Activity,
    session: SessionName,
    sessionsClient: SessionsClient,
    queryInput: QueryInput
) :
    AsyncTask<Void?, Void?, DetectIntentResponse?>() {
    private val session: SessionName
    private val sessionsClient: SessionsClient
    private val queryInput: QueryInput
    override fun doInBackground(vararg params: Void?): DetectIntentResponse? {
        try {
            val detectIntentRequest: DetectIntentRequest = DetectIntentRequest.newBuilder()
                .setSession(session.toString())
                .setQueryInput(queryInput)
                .build()
            return sessionsClient.detectIntent(detectIntentRequest)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return null
    }

    override fun onPostExecute(response: DetectIntentResponse?) {
        (activity as ChatActivity).callbackV2(response)
    }

    init {
        this.session = session
        this.sessionsClient = sessionsClient
        this.queryInput = queryInput
    }
}